application/include/mmul.hpp

```diff
@@ -17,15 +17,14 @@
 using namespace Eigen;
 
 template <typename Executor,
-          typename execution::executor_of_type<inline_executor, Executor> = 0>
+          typename execution::instance_of<inline_executor, Executor> = 0>
```

Comment:
Rename to `B_is_instance_of_A` or something else which grammatically suggests that the second argument is instantiated template of arg 1

Reply:
How does `is_instance_of_base` sound? It's on the lines of `is_base_of`?